### PR TITLE
PTRENG-3249, RT provider, fix keypair tests failure. Removed username…

### PR DESCRIPTION
**Pull request by danielmkn**
… and access key conflict. Update RT version in the script. Updated documentation

### CODE REVIEW

File: docs/resources/artifactory_keypair.md [hunk 1]
```diff
@@ -36,3 +42,6 @@ The following arguments are supported:
 * `passphrase` - (Optional/Questionable)  - This will be used to decrypt the private key. Validated server side.
 * `public_key` - (Required)  - duh! This will have it's pem format validated
 * `unavailable` - (Computed) - it's unknown what this does, but, it's returned in the payload and there is no known place to set it in the UI
+
+Artifactory REST API call Get Key Pair doesn't return keys `private_key` and `passphrase`, but consumes these keys in the POST call.
+The meta-argument `lifecycle` used here to make Provider ignore the verification for these two keys in the Terraform state.
```
Comment (alexhung) [inline docs/resources/artifactory_keypair.md:47]: s/verification/changes

File: docs/resources/artifactory_local_alpine_repository.md [hunk 2]
```diff
@@ -11,6 +11,12 @@ resource "artifactory_keypair" "some-keypairRSA" {
...
   public_key  = file("samples/rsa.pub")
+  lifecycle {
+    ignore_changes = [
+      private_key,
+      passphrase,
+    ]
+  }
 }
 
 resource "artifactory_local_alpine_repository" "terraform-local-test-alpine-repo-basic" {
```
Comment (alexhung) [inline docs/resources/artifactory_local_alpine_repository.md:22]: Missing the new comment like in `artifactory_keypair.md`

File: docs/resources/artifactory_local_debian_repository.md [hunk 3]
```diff
@@ -11,13 +11,25 @@ resource "artifactory_keypair" "some-keypairGPG1" {
...
   pair_type   = "GPG"
   alias       = "foo-alias2"
   private_key = file("samples/gpg.priv")
   public_key  = file("samples/gpg.pub")
+  lifecycle {
+    ignore_changes = [
+      private_key,
+      passphrase,
+    ]
+  }
```
Comment (alexhung) [inline docs/resources/artifactory_local_debian_repository.md:32]: Same for this file

File: pkg/artifactory/resource_artifactory_keypair.go [hunk 4]
```diff
@@ -61,15 +61,16 @@ func resourceArtifactoryKeyPair() *schema.Resource {
...
 			},
 			"passphrase": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Description: "Used to decrypt the private key (if applicable). Will be verified server side",
-				ForceNew:    true,
+				Type:             schema.TypeString,
+				Optional:         true,
+				DiffSuppressFunc: ignoreEmpty,
+				Description:      "Used to decrypt the private key (if applicable). Will be verified server side",
```
Comment (alexhung) [inline pkg/artifactory/resource_artifactory_keypair.go:72]: We should add to this (and `private_key`) that RT doesn't return the value after creation.

File: scripts/run-artifactory.sh [hunk 5]
```diff
@@ -5,7 +5,7 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" > /dev/null && pwd )"
 set -euf
 
 docker run -i -t -d --rm -v "${SCRIPT_DIR}/artifactory.lic:/artifactory_extra_conf/artifactory.lic:ro" \
-  -p8081:8081 -p8082:8082 -p8080:8080 releases-docker.jfrog.io/jfrog/artifactory-pro:7.24.3  #7.27.6  #7.27.10 #7.24.3
+  -p8081:8081 -p8082:8082 -p8080:8080 releases-docker.jfrog.io/jfrog/artifactory-pro:7.27.10 #7.24.3
```
Comment (alexhung) [inline scripts/run-artifactory.sh:8]: Please remove the comment at the end of the line

File: pkg/artifactory/provider.go [hunk 6]
```diff
@@ -25,12 +25,11 @@ func Provider() *schema.Provider {
 				ValidateFunc: validation.IsURLWithHTTPorHTTPS,
 			},
 			"username": {
-				Type:          schema.TypeString,
-				Optional:      true,
-				DefaultFunc:   schema.EnvDefaultFunc("ARTIFACTORY_USERNAME", nil),
-				ConflictsWith: []string{"access_token", "api_key"},
```
Comment (alexhung) [inline pkg/artifactory/provider.go:31]: Does `username ` also conflict with `api_key`? Or just `access_token`?

**Review by alexhung — CHANGES_REQUESTED** [review]

File: docs/resources/artifactory_local_alpine_repository.md (hunk 2 above)
Comment (danielmkn) [reply docs/resources/artifactory_local_alpine_repository.md:22]: do we need this comment for every repo doc?

File: scripts/run-artifactory.sh (hunk 5 above)
Comment (danielmkn) [reply scripts/run-artifactory.sh:8]: done

File: pkg/artifactory/provider.go (hunk 6 above)
Comment (danielmkn) [reply pkg/artifactory/provider.go:31]: ConflictsWith defines if it conflicts with username. So, by removing the field entirely, we solve this problem.

File: docs/resources/artifactory_keypair.md (hunk 1 above)
Comment (danielmkn) [reply docs/resources/artifactory_keypair.md:47]: done

File: docs/resources/artifactory_local_alpine_repository.md (hunk 2 above)
Comment (alexhung) [reply docs/resources/artifactory_local_alpine_repository.md:22]: Yes. Until we have a script to auto generate these docs, we should keep them all consistent.

File: pkg/artifactory/provider.go (hunk 6 above)
Comment (alexhung) [reply pkg/artifactory/provider.go:31]: That's not what this line says though. It says "username field conflicts with both access_token field and api_key field". So if access_token field is the one in combination with username causing issue, we should still include api_key in this line.

Comment (danielmkn) [reply pkg/artifactory/provider.go:31]: testing with api_key

File: pkg/artifactory/resource_artifactory_keypair.go (hunk 4 above)
Comment (danielmkn) [reply pkg/artifactory/resource_artifactory_keypair.go:72]: added to both

**Review by alexhung — APPROVED** [review]

**Pull request merged**